src/irmin-pack/s.ml

```diff
@@ -55,7 +55,34 @@ module type Specifics = sig
   (** [flush t] flush read-write pack on disk. Raises [RO_Not_Allowed] if called
       by a readonly instance.*)
 
-  val gc : ?unlink:bool -> repo -> commit_key -> unit
+  val start_gc :
+    ?unlink:bool ->
+    ?auto_finalise:bool ->
+    throttle:[ `Block | `Skip ] ->
+    repo ->
+    commit_key ->
+    bool Lwt.t
+  (** [start_gc] tries to start the gc process and returns true if the gc is
+      launched.
+
+      If [unlink] is false then temporary files on disk are kept after the gc
+      finished. This option is useful for debugging. The default is true.
+
+      If [auto_finalise] is true (the default) then there is no need to call
+      explicitly [finalise_gc].
+
+      If [throttle] is [Skip] and there is a concurrent gc, [start_gc] returns
+      false immediately without launching a second gc. If [throttle] is [Block]
+      and there is a concurrent gc, [start_gc] blocks waiting for the previous
+      gc to finish and only after launches a second gc. *)
+
+  val finalise_gc : ?wait:bool -> repo -> bool Lwt.t
+  (** [finalise_gc unlink wait gen pid repo] waits for the gc process to finish
+      finalises the gc (swaps files and clears caches). It needs the generation
+      and pid returned by the call to [start_gc].
+
+      If [wait] is true it blocks waiting for the gc process to finish. By
+      default, it is false and the call returns immediately. *)
```

Comment:
[finalise_gc ?wait repo] waits for the gc process to finish in order to finalise it. It returns true if a GC was finalised.

Finalising consists of mutating [repo] so that it points to the new file and to flush the internal caches that could be referencing GCed objects.

If [wait = true] (the default), the call blocks until the GC process finishes, otherwise it doesn't block.

TODO: Detail exceptions raised